File: src/test_files/typescript/MultiClassFixture.ts
// src/test_files/typescript/MultiClassFixture.ts
//
// Multi-class TypeScript regression fixture.
//
// Tests the per-class metadata invariant:
//   A meta-layer may inspect only the exact source span belonging to the
//   type it is enriching. It must never infer ownership from neighboring
//   or whole-file text.
//
// This file contains multiple classes with different Angular decorators
// and plain classes interspersed. Each class's markers must be emitted
// ONLY for that class, never for another class.

import { Component, Injectable, Input, Output, EventEmitter } from '@angular/core';

// ════════════════════════════════════════════════════════════════════════
// Class 1: @Component with selector 'app-hello'
// Expected: Φcmp:HelloComponent sel=app-hello
// ════════════════════════════════════════════════════════════════════════
@Component({
    selector: 'app-hello',
    template: '<h1>Hello</h1>'
})
export class HelloComponent {
    @Input() name: string = '';
}

// ════════════════════════════════════════════════════════════════════════
// Class 2: Plain class — NO Angular decorator
// Expected: NO Φ markers of any kind
// ════════════════════════════════════════════════════════════════════════
export class DataModel {
    constructor(public id: number, public label: string) {}
}

// ════════════════════════════════════════════════════════════════════════
// Class 3: @Injectable service
// Expected: Φsvc:DataService scope=root
// CRITICAL: Must NOT inherit HelloComponent's Φcmp marker
// ════════════════════════════════════════════════════════════════════════
@Injectable({ providedIn: 'root' })
export class DataService {
    fetchAll(): Promise<DataModel[]> { return Promise.resolve([]); }
}

// ════════════════════════════════════════════════════════════════════════
// Class 4: @Component with DIFFERENT selector 'app-goodbye'
// Expected: Φcmp:GoodbyeComponent sel=app-goodbye
// CRITICAL: Must NOT inherit HelloComponent's 'app-hello' or DataService's Φsvc
// ════════════════════════════════════════════════════════════════════════
@Component({
    selector: 'app-goodbye',
    template: '<h1>Goodbye</h1>'
})
export class GoodbyeComponent {
    @Output() closed = new EventEmitter<void>();
}

// ════════════════════════════════════════════════════════════════════════
// Class 5: Plain interface — NO Angular decorator
// Expected: NO Φ markers
// ════════════════════════════════════════════════════════════════════════
export interface ApiResponse {
    success: boolean;
    data: unknown;
}

File: src/test_files/dotnet/MultiClassFixture.cs
// src/test_files/dotnet/MultiClassFixture.cs
//
// Multi-class C# / .NET regression fixture.
//
// Tests the per-class metadata invariant:
//   A meta-layer may inspect only the exact source span belonging to the
//   type it is enriching. It must never infer ownership from neighboring
//   or whole-file text.
//
// This file contains multiple classes with different .NET framework patterns
// and plain classes interspersed. Each class's markers must be emitted
// ONLY for that class, never for another class.

using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using Microsoft.AspNetCore.SignalR;
using System.Threading.Tasks;

namespace CleanCtx.TestFiles.Dotnet.MultiClass;

// ════════════════════════════════════════════════════════════════════════
// Class 1: ASP.NET Core Controller with [ApiController] and [Route]
// Expected: Φ marker with ProductsController
// ════════════════════════════════════════════════════════════════════════
[ApiController]
[Route("api/[controller]")]
public class ProductsController : ControllerBase
{
    [HttpGet]
    public IActionResult GetAll()
    {
        return Ok(new[] { "Product1", "Product2" });
    }
}

// ════════════════════════════════════════════════════════════════════════
// Class 2: Plain POCO — NO framework attributes
// Expected: NO Φ markers
// ════════════════════════════════════════════════════════════════════════
public class Product
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
}

// ════════════════════════════════════════════════════════════════════════
// Class 3: SignalR Hub
// Expected: Φ marker with NotificationHub
// CRITICAL: Must NOT inherit ProductsController's ASP.NET markers
// ════════════════════════════════════════════════════════════════════════
public class NotificationHub : Hub
{
    public async Task SendMessage(string user, string message)
    {
        await Clients.All.SendAsync("ReceiveMessage", user, message);
    }
}

// ════════════════════════════════════════════════════════════════════════
// Class 4: EF Core DbContext
// Expected: Φ marker with InventoryDbContext
// CRITICAL: Must NOT inherit ProductsController's [ApiController] markers
// ════════════════════════════════════════════════════════════════════════
public class InventoryDbContext : DbContext
{
    public DbSet<Product> Products { get; set; } = null!;
}

// ════════════════════════════════════════════════════════════════════════
// Class 5: Plain POCO — NO framework attributes
// Expected: NO Φ markers
// ════════════════════════════════════════════════════════════════════════
public class InventoryItem
{
    public int Id { get; set; }
    public int Quantity { get; set; }
}